### ISSUE TITLE
Plot not showing when NAs in matrix

### ISSUE THREAD

**Issue by juferban** [Alanocallaghan/d3heatmap#3]
Hi Alan,

I noticed that for cases where I have a matrix with some Na values the heatmap does not get displayed.

I was looking at your code and I think it has to do with the scalecolors not having an na.color attached like in the original version but I cannot confirm this. (around line 380)

Here is some code to replicate the error:
```
library(shiny)
library(d3heatmap)

x <- matrix( c(2, 4, NA, 8,5 ,9 ,3 ,5 ,7 ,9 ,NA ,5 ,3 ,6 ,7 , 1, 5, 7, 6, 1),
             nrow=4, ncol=5)

ui <- fluidPage(
  titlePanel("d3heatmap example"),  
  d3heatmapOutput("multiHeatmap", height = "800px") 
)

server <- function(input, output) {
  output$multiHeatmap <- renderD3heatmap({
    d3heatmap(x, 
              main = "TEST",
              show_color_legend = TRUE,
              scale = "column",
              na.color = 'grey'
    )           
  })
}
  
shinyApp(ui = ui, server = server)
```

Thanks,

Julio

**Comment by juferban** [comment]
Hi Alan,

I've been doing some more testing trying to find out what causes the matrix to show all blank when there are NA values. I tried commenting most of your new code from the d3heatmap.R file with the idea to keep adding section back one by one and try to find the problematic one. 
Unfortunatelly commenting the new code in that file didn't solve the problem which may mean that the issue is coming from somewhere else, maybe the css file.
I'll try to investigate a bit more and will let you know if I find anything. 

Again thanks for you work improving this library.

Julio

**Comment by Alanocallaghan** [comment]
Hi Julio, I'll look into it this weekend. Apologies for the issue. It's most likely coming from the JS code but there is a small chance it's caused by R

**Comment by Alanocallaghan** [comment]
Hi Julio,

I have pushed a quick change that should fix this for now. Thanks again for the detailed issue and the MWE, I'll try to investigate this properly soon

**Comment by Alanocallaghan** [comment]
The parameter `na_color` should allow you to choose what color NA values have in the heatmaps. They will show up on the mouseover as "null" which is an unfortunate difference between how R and JS handle missing values. Ideally I would like to resolve that as NA, NaN and NULL are not equivalent.

**Comment by juferban** [comment]
Hi Alan,

Thanks a lot for you quick work on this, I really appreciate it.

I downloaded and tested your latest version and now the plot works. I was not able to change the  NA colors though.
I did a quick search on you code and notice that  in inst/htmlwidgets/lib/d3heatmapcore/heatmapcore.js
you have added 
  
`opts.na_color = options.NA_color; in line 203.`

I think there is a typo there and the second NA needs to be also lower case (as you use lower case in the option argument in the R file). I changed it in my local install for testing and it worked on changing the color. 
` opts.na_color = options.na_color;`

Once again big thanks to you.

Julio
